### PR TITLE
Fix test edge cases

### PR DESCRIPTION
**Pull request by killercup**
Noticed this while adding test case for ~~fun~~ https://github.com/rust-lang-nursery/rust-clippy/pull/2350

### CODE REVIEW

File: tests/everything.rs [hunk 1]
```diff
@@ -143,7 +143,7 @@ fn test_rustfix_with_file<P: AsRef<Path>>(file: P) -> Result<(), Box<Error>> {
 
     let mut fixed = code.clone();
 
-    for sug in suggestions {
+    for sug in suggestions.into_iter().rev() {
```
Comment (oli-obk) [inline tests/everything.rs]: needs a test

Comment (killercup) [reply tests/everything.rs]: Only possible with an additional test that has a replacement that condenses n lines into m<n  lines followed by ≥1 replacements.

…which I just added

**Comment by oli-obk** [comment]
now we know that a missing review will even fail bors. Although the silent failure is suboptimal

**Comment by killercup** [comment]
Hah, thanks for testing ;)

What was the issue? That it couldn't merge the PR because Github rejected its request?

Do you want me to change the settings to only require bors but not a github review?

**Comment by oli-obk** [comment]
> Do you want me to change the settings to only require bors but not a github review?

nah, I'm a quick learner

**Comment by killercup** [comment]
Me: the fool who set this up

You: a quick learner

Me in a month: absolutely clueless why bors doesn't work
